src/codegen/operator/table_scan_translator.cpp

```diff
@@ -47,22 +56,23 @@ TableScanTranslator::TableScanTranslator(const planner::SeqScanPlan &scan,
       pipeline.InstallBoundaryAtOutput(this);
     }
   }
-
-  auto &codegen = GetCodeGen();
   auto &runtime_state = context.GetRuntimeState();
   selection_vector_id_ = runtime_state.RegisterState(
       "scanSelVec",
       codegen.ArrayType(codegen.Int32Type(), Vector::kDefaultVectorSize), true);
 
-  LOG_TRACE("Finished constructing TableScanTranslator ...");
+  storage::ZoneMapManager *zone_map_manager =
+      storage::ZoneMapManager::GetInstance();
+  zone_map_table_exists = zone_map_manager->ZoneMapTableExists();
```

Comment:
Is this check needed here? Is there a case where the zone map table wouldn't exist? If so, what would do you do?

Reply:
Ah, since `ZoneMapManager` is a singleton, can you just initialize it in `main`?

Reply:
Are you talking about initializing zone_map_table_exists and I also didnt understand which main you are referring to ?

Reply:
I guess the question is would the zone map table not exist at this point?